src/utils/permissions.ts

```diff
@@ -0,0 +1,27 @@
+import { CommandInteraction, PermissionString, TextChannel } from 'discord.js';
+import { check } from './emotes.json';
+
+export function hasPermissions(
+	permissions: PermissionString[],
+	interaction: CommandInteraction,
+): boolean {
+	if (!permissions) return true;
+
+	const { user, channel } = interaction;
+	const c = channel as TextChannel;
+
+	const permissionsBooleans = permissions.map(p => !c.permissionsFor(user)!.has(p));
+	const missingPermissions = (permissions.filter((p, i) => !permissionsBooleans[i])).map(p => `\`${p}\``);
+
+	if (permissionsBooleans.includes(false)) {
+		interaction.reply([
```

Comment:
Move this to messages aswell

Reply:
Yeah, sure
I just made some of the messages in `messages.json`, hopefully today i will make it to all of them